src/util/TchapUtils.ts

```diff
@@ -26,19 +26,40 @@ export default class TchapUtils {
     /**
      * For the current user, get the room federation options.
      *
-     * @returns { showRoomFederationOption: boolean, roomFederationDefault: boolean } options
+     * @returns { showRoomFederationOption: boolean,roomDefaultFederation: { external: boolean, private: boolean, forum: boolean } } options
      */
-    static getRoomFederationOptions(): { showRoomFederationOption: boolean, roomFederationDefault: boolean } {
+    public static getRoomFederationOptions(): {
+        showRoomFederationOption: boolean,
+        roomDefaultFederation: {
+            external: boolean,
+            private: boolean,
+            forum: boolean
+        }
+    } {
         const cli = MatrixClientPeg.get();
         const baseDomain = cli.getDomain();
 
         // Only show the federate switch to defense users : it's difficult to understand, so we avoid
         // displaying it unless it's really necessary.
         if (baseDomain === 'agent.intradef.tchap.gouv.fr') {
-            return { showRoomFederationOption: true, roomFederationDefault: false };
```

Comment:
it should be : {showForumFederationSwitch: true, forumFederationSwitchDefaultValue: false}

Reply:
Modified